### PR TITLE
[TEST] First few test for the user_bin_sequence class

### PR DESCRIPTION
**Pull request by smehringer**
Doesn't increase the coverage yet I think. 
But good to have anyway.

### CODE REVIEW

File: test/api/union/user_bin_sequence_test.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#include <gtest/gtest.h>
+
+#include <unordered_set>
+
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:4]:
```suggestion
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#include <chopper/union/user_bin_sequence.hpp>
+
+#include "../api_test.hpp"
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:7]:
```suggestion
#include <gtest/gtest.h>

#include <chopper/union/user_bin_sequence.hpp>

#include "../api_test.hpp"
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+// inherits from user_bin_sequence to test private members
+struct user_bin_sequence_test : public ::testing::Test, public user_bin_sequence
+{
+public:
+
+    std::vector<std::string> global_filenames{"small.fa", "small.fa", "small2.fa", "small2.fa"};
+    std::vector<size_t> global_kmer_counts{500, 600, 700, 800};
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:13]: Could these be private?

File: test/api/union/user_bin_sequence_test.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    std::vector<std::string> const & get_filenames()
+    {
+        return *this->filenames;
+    }
+
+    std::vector<size_t> const & get_user_bin_kmer_counts()
+    {
+        return *this->user_bin_kmer_counts;
+    }
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:23]:
```suggestion
    std::vector<std::string> const & get_filenames() const
    {
        return *this->filenames;
    }

    std::vector<size_t> const & get_user_bin_kmer_counts() const
    {
        return *this->user_bin_kmer_counts;
    }
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+TEST_F(user_bin_sequence_test, construction)
+{
+    EXPECT_TRUE(std::is_default_constructible<user_bin_sequence>::value);
+    EXPECT_TRUE(std::is_copy_constructible<user_bin_sequence>::value);
+    EXPECT_TRUE(std::is_move_constructible<user_bin_sequence>::value);
+    EXPECT_TRUE(std::is_destructible<user_bin_sequence>::value);
+
+    EXPECT_FALSE(std::is_copy_assignable<user_bin_sequence>::value); // class has a const pointer
+    EXPECT_FALSE(std::is_move_assignable<user_bin_sequence>::value); // class has a const pointer
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:40]:
```suggestion

    // class has a const pointer
    EXPECT_FALSE(std::is_copy_assignable<user_bin_sequence>::value);
    EXPECT_FALSE(std::is_move_assignable<user_bin_sequence>::value); 
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    std::vector<size_t> kmer_counts{500, 500};
+    user_bin_sequence ubs{filenames, kmer_counts};
+}
+
+TEST_F(user_bin_sequence_test, apply_permutation)
+{
+    std::vector<size_t> const permutation{2, 1, 3, 0};
+
+    EXPECT_RANGE_EQ(this->get_filenames(), global_filenames);
+    EXPECT_RANGE_EQ(this->get_user_bin_kmer_counts(), global_kmer_counts);
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:53]: These objects should have the same address, no?

File: test/api/union/user_bin_sequence_test.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    this->apply_permutation(permutation);
+
+    EXPECT_RANGE_EQ(this->get_filenames(), (std::vector<std::string>{"small2.fa", "small.fa", "small2.fa", "small.fa"}));
+    EXPECT_RANGE_EQ(this->get_user_bin_kmer_counts(), (std::vector<size_t>{700, 600, 800, 500}));
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:58]: The global ones should also have changed?

**Review by eseiler — APPROVED** [review]
Some comments for the next PR

**Comment by smehringer** [comment]
why are you merging when you have several valid remarks?

**Pull request merged**